apps/emqx_utils/src/emqx_utils_stream.erl

```diff
@@ -190,7 +190,7 @@ transpose_tail(S, Tail) ->
 %% @doc Make a stream by concatenating multiple streams.
 -spec chain([stream(X)]) -> stream(X).
 chain(L) ->
-    lists:foldl(fun chain/2, empty(), L).
+    lists:foldr(fun chain/2, empty(), L).
```

Comment:
What is the reason of this change?

Reply:
This preserves the order of `L`, as evidenced by some of the `emqx_ds` tests. Without this change, once `add_generation` events are added into a test event stream, [this call](https://github.com/emqx/emqx/blob/a37376e3530aa10b141ffb9b0e95c0149452379d/apps/emqx_durable_storage/test/emqx_ds_test_helpers.erl#L287-L288) results in a stream which goes from newer generations to older.

Reply:
Thanks!